### PR TITLE
COG-815 User Detail Page appears to save changes even if error is presented when attempting to activate a user with invalid CWS Login

### PR DESCRIPTION
**Pull request by Nasir2712**
#### Which User story does this relate to (link)?
-https://osi-cwds.atlassian.net/browse/COG-815

#### Brief feature description
-After hitting save when in edit mode, if there is an error, the view should not switch to show mode and the changes should revert back to what they were before hitting save.

### CODE REVIEW

File: app/javascript/reducers/detailsReducers.js [hunk 1]
```diff
@@ -14,7 +17,9 @@ function fetchDetails(state = { details: null, fetching: false }, action) {
         ...state,
         fetching: false,
         details: userRecords,
+        originalDetails: userRecords,
```
[the remark this answers is not in the document]
Comment (RamuKammagani-osi) [reply app/javascript/reducers/detailsReducers.js]: I don't see originalDetails were used in container so may i know the purpose of adding it??

[the remark this answers is not in the document]
Comment (Nasir2712) [reply app/javascript/reducers/detailsReducers.js]: it's kinda complicated, but I will try my best to explain it. Let's say user changed the status from inactive to active, changed the role from cws-worker to office administrator and clicked save. The user got an error, the purpose of adding original details is to reset all the changes back to how it was, before clicking edit. This is kinda an indication to the user saying, the changes were not saved. Think of it as a extra row in the database, whose purpose is to just save the details when there were loaded.

[the remark this answers is not in the document]
Comment (Nasir2712) [reply app/javascript/reducers/detailsReducers.js]: Also having said that, we might not need it based on the requirements of the story. I'll update this in a few.

[the remark this answers is not in the document]
Comment (Nasir2712) [reply app/javascript/reducers/detailsReducers.js]: Had the discussion with Leon, got it removed. Feel free to review again. Thanks!

File: app/javascript/actions/detailActions.js [hunk 2]
```diff
@@ -3,13 +3,19 @@ import {
...
   HANDLE_DROPDOWN_CHANGE,
+  HANDLE_EDIT_BUTTON_CHANGE,
 } from './actionTypes';
 
 export const handleDropdownChangeAction = (name, value) => ({
   type: HANDLE_DROPDOWN_CHANGE,
   payload: { name, value },
 });
 
+export const handleEditButtonChangeAction = value => ({
```
Comment (kelfink) [inline app/javascript/actions/detailActions.js]: There should be a unit test for this

Comment (Nasir2712) [reply app/javascript/actions/detailActions.js]: good catch

Comment (Nasir2712) [reply app/javascript/actions/detailActions.js]: updated

**Pull request merged**